content/en/tests/setup/python.md

```diff
@@ -163,6 +163,213 @@ assert True
+
+Call `ddtrace.ext.test_visibility.api.TestSuiteId()`, providing the parent module's `TestModuleId`, and the suite's name, as arguments, to create a `TestSuiteId`.
+
+Call `ddtrace.ext.test_visibility.api.TestSuite.discover()`, passing the `TestSuiteId` object as an argument, to discover the suite.
+
+Call `ddtrace.ext.test_visibility.api.TestSuite.start()`, passing the `TestSuiteId` object as an argument, to start the suite.
+
+After all the children items within the suite have completed, call `ddtrace.ext.test_visibility.api.TestSuite.finish()`, passing the `TestSuiteId` object as an argument.
+
+##### Test
+
+A test represents a single test case that is executed as part of a test suite.
+
+Call `ddtrace.ext.test_visibility.api.TestId()`, providing the parent suite's `TestSuiteId`, and the test's name, as arguments to create a `TestId`. The `TestId()` method accepts a JSON-parseable string as the optional `parameters` argument. The `parameters` argument can be used to distinguish parametrized tests that have the same name, but different parameter values.
+
+Call `ddtrace.ext.test_visibility.api.Test.discover()`, passing the `TestId` object as an argument, to discover the test. The `Test.discover()` classmethod accepts a string as the optional `resource` parameter, which default to the `TestId`'s `name`.
+
+Call `ddtrace.ext.test_visibility.api.Test.start()`, passing the `TestId` object as an argument, to start the test.
+
+Call `ddtrace.ext.test_visibility.api.Test.mark_pass()`, passing the `TestId` object as an argument, to mark that the test has passed successfully.
+Call `ddtrace.ext.test_visibility.api.Test.mark_fail()`, passing the `TestId` object as an argument, to mark that the test has failed. `mark_fail()` accepts an optional `TestExcInfo` object as the `exc_info` parameter.
+Call `ddtrace.ext.test_visibility.api.Test.mark_skip()`, passing the `TestId` object as an argument, to mark that the test was skipped. `mark_skip()` accepts an optional string as the `skip_reason` parameter.
+
+###### Exception information
+
+The `ddtrace.ext.test_visibility.api.Test.mark_fail()` classmethod holds information about exceptions encountered during a test's failure.
+
+The `ddtrace.ext.test_visibility.api.TestExcInfo()` method takes three positional parameters:
+- `exc_type`: the type of the exception encountered
+- `exc_value`: the `BaseException` object for the exception
+- `exc_traceback`: the `Traceback` object for the exception
+
+###### Codeowner information
+
+The `ddtrace.ext.test_visibility.api.Test.discover()` classmethod accepts an optional list of strings as the `codeowners` parameter.
+
+###### Test source file information
+
+The `ddtrace.ext.test_visibility.api.Test.discover()` classmethod accepts an optional `TestSourceFileInfo` object as the `source_file_info` parameter. A `TestSourceFileInfo` object represents the path and, optionally, the start and end lines for a given test.
```

Comment:
```suggestion
The `ddtrace.ext.test_visibility.api.Test.discover()` classmethod accepts an optional `TestSourceFileInfo` object as the `source_file_info` parameter. A `TestSourceFileInfo` object represents the path and optionally, the start and end lines for a given test.
```

Reply:
@urseberry , just double-checking this one. It kinda looks weird to me that there's just the one comma after "optionally".